flink/flink-common/src/main/java/com/netease/arctic/flink/FlinkSchemaUtil.java

```diff
@@ -197,4 +242,191 @@ public static void addPrimaryKey(
+   *
+   * @param schema Flink TableSchema.
+   * @return tableProperties.
+   */
+  public static Map<String, String> generateExtraOptionsFrom(TableSchema schema) {
+    Map<String, String> properties = Maps.newHashMap();
+
+    // add properties for computeColumns
+    Map<String, String> computeColumnProperties = serializeComputeColumn(schema);
+    properties.putAll(computeColumnProperties);
+
+    // add properties for watermark,only support one watermark now
+    List<WatermarkSpec> watermarkSpecs = schema.getWatermarkSpecs();
+    if (!watermarkSpecs.isEmpty()) {
+      if (watermarkSpecs.size() > 1) {
+        throw new IllegalStateException("Multiple watermark definition is not supported yet.");
+      }
+      properties.putAll(serializeWatermarkSpec(watermarkSpecs.get(0)));
+    }
+
+    return properties;
+  }
+
+  /** serialize computeColumns into properties */
+  private static Map<String, String> serializeComputeColumn(TableSchema schema) {
+    Map<String, String> serialized = new HashMap<>();
+    List<TableColumn> tableColumns = schema.getTableColumns();
+    // index in compute Column, starting from 1
+    int computeIndex = 1;
+    for (TableColumn column : tableColumns) {
+      if (column instanceof ComputedColumn) {
+        serialized.put(
+            compoundKey(FLINK_PREFIX, COMPUTED_COLUMNS, computeIndex, NAME), column.getName());
+        serialized.put(
+            compoundKey(FLINK_PREFIX, COMPUTED_COLUMNS, computeIndex, DATA_TYPE),
+            column.getType().getLogicalType().asSerializableString());
+        serialized.put(
+            compoundKey(FLINK_PREFIX, COMPUTED_COLUMNS, computeIndex, EXPR),
+            ((TableColumn.ComputedColumn) column).getExpression());
+        computeIndex++;
+      }
+    }
+    return serialized;
+  }
+
+  /** deserialize computeColumns from properties */
+  private static TableColumn deserializeComputeColumn(
+      Map<String, String> tableProperties, int index, List<String> fieldNames) {
+    String expr = tableProperties.get(compoundKey(FLINK_PREFIX, COMPUTED_COLUMNS, index, EXPR));
+    if (!isExprContainField(expr, fieldNames)) {
+      throw new IllegalStateException(
+          "expression " + expr + " do not match any columns in amoro. ");
+    }
+    DataType dataType =
+        TypeConversions.fromLogicalToDataType(
+            LogicalTypeParser.parse(
+                tableProperties.get(
+                    compoundKey(FLINK_PREFIX, COMPUTED_COLUMNS, index, DATA_TYPE))));
+    TableColumn column =
+        TableColumn.computed(
+            tableProperties.get(compoundKey(FLINK_PREFIX, COMPUTED_COLUMNS, index, NAME)),
+            dataType,
+            expr);
+    return column;
+  }
+
+  private static boolean isExprContainField(String expr, List<String> fieldNames) {
+    if (expr.equalsIgnoreCase(PROCTIME_FUNCTION)) {
+      return true;
+    }
+    for (String fieldName : fieldNames) {
+      if (expr.contains("`" + fieldName + "`")) {
```

Comment:
Is charactor ` always required in expr?

Reply:
> Is charactor ` always required in expr?

charactor ` will be keeped in property for expresion during deserialize compute column from Flink TableSchema.

considering the situation below
```         
arctic_catalog.test_db.test_table( Id INT);
```
properity
```
flink.schema.column.1.expr=`oldId`+5
```
whitout charactor `,  compute column for index  1 would be deserialized into compute colum  mistakenly.